webapp/lib/platform.dart

```diff
@@ -138,13 +141,14 @@ Future updateUnread(List<Conversation> conversations, bool newValue) async {
       : "${conversations.length} conversations"
   }");
   if (conversations.isEmpty) return null;
-  var batch = _firestoreInstance.batch();
+  var batch = FirestoreDocUpdateBatch(_firestoreInstance.batch());
   int batchSize = 0;
   for (var conversation in conversations) {
     conversation.updateUnread(_firestoreInstance, conversation.documentPath, newValue, batch);
+    ++batchSize;
     if (batchSize == _MAX_BATCH_SIZE) {
       await batch.commit();
-      batch = _firestoreInstance.batch();
+      batch = FirestoreDocUpdateBatch(_firestoreInstance.batch());
       batchSize = 0;
     }
```

Comment:
I think we also need to commit the final set after the user interaction - the problem is things like what happenes if we do a bunch of stuff and then close the window. We have to commit all of the outstanding batched before the application closes. 

Given typical write performance on Firebase, I think we can be pretty eager about calling commit

Reply:
Done.